### PR TITLE
Add Ruby 3.0 support

### PR DESCRIPTION
**Pull request by mbj**

### CODE REVIEW

File: mutant.sh [hunk 1]
```diff
@@ -1,11 +1,12 @@
...
-  --ignore-subject Mutant::CLI#add_debug_options                 \
-  --ignore-subject Mutant::Isolation::Fork::Parent#call          \
-  --ignore-subject Mutant::Mutator::Node::Argument#skip?         \
-  --ignore-subject Mutant::Mutator::Node::ProcargZero#dispatch   \
-  --ignore-subject Mutant::Mutator::Node::When#mutate_conditions \
-  --ignore-subject Mutant::Zombifier#call                        \
+bundle exec mutant run                                                 \
+  --zombie                                                             \
+  --ignore-subject Mutant::CLI#add_debug_options                       \
+  --ignore-subject Mutant::Expression::Namespace::Recursive#initialize \
```
Comment (mbj) [inline mutant.sh:6]: A mutation for 3.0 specific code here. We need version specific configuration to exclude the subject (or mutation) on specific versions.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -119,7 +119,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        ruby: [ruby-2.7]
+        ruby: [ruby-2.7, ruby-3.0]
```
Comment (dgollahon) [inline .github/workflows/ci.yml:122]: Are you intentionally running rubocop on two versions of ruby?

File: lib/mutant/expression/namespace.rb [hunk 3]
```diff
@@ -17,7 +17,7 @@ class Recursive < self
         def initialize(*)
           super
 
-          @syntax = "#{scope_name}*"
+          @syntax = "#{scope_name}*".freeze # rubocop:disable Style/RedundantFreeze
```
Comment (dgollahon) [inline lib/mutant/expression/namespace.rb:20]: Also curious if you know something I don't about this--I thought this was automatically frozen even with interpolation?

```ruby
# frozen_string_literal: true

class Foo
  def initialize
    @test = "#{rand}-test"
    pp @test.frozen?
  end
end

Foo.new # outputs `true`
```

File: .github/workflows/ci.yml (hunk 2 above)
Comment (mbj) [reply .github/workflows/ci.yml:122]: for the moment yes, will soon remove 2.7.

File: lib/mutant/expression/namespace.rb (hunk 3 above)
Comment (mbj) [reply lib/mutant/expression/namespace.rb:20]: https://twitter.com/_m_b_j_/status/1342635659547340800

TLDR: MRI decided that, as each dstr gets allocated always freezing makes no sense.

I disagree as freezing is also a correctness mechanism, not just a memory optimization.

Instead: They should have patched MRI such as `+"some-#{dstr}"` would create a mutable DSTR, without frozen intermediary.

Comment (dgollahon) [reply lib/mutant/expression/namespace.rb:20]: ...oh. I don't like that at all. :(
